server/http/protolet/protolet.go

```diff
@@ -78,7 +116,14 @@ func WriteProtoToResponse(rsp proto.Message, w http.ResponseWriter, r *http.Requ
 	}
 }
 
-func GenerateHTTPHandlers(server interface{}) (http.HandlerFunc, error) {
+type HTTPHandlers struct {
```

Comment:
I.... can't say I really love this but it does work and that's worth a lot 👍 

If we were doing this again what I would do is -- have protolet just make and run a self-RPC (like, RPC to localhost), rather than calling the wrapped function directly. That way the protolet can just live in unauthenticated space, call the RPC, then your auth code runs, then the RPC fires, and bob's your uncle.

Anyway, I'll settle for a TODO(tylerw) in here to "make protolet self-RPC" or something. That work?

Also I'm really impressed you figured all this reflection crap out, it's fairly gnarly :) Nice work on that!

Reply:
Lol, yeah I definitely think the self-RPC pattern would be pretty clean, so I added a TODO. Just not sure how much work that would be, since HTTP headers/cookies etc. are no longer available when you're responding to the self-RPC, and I'm not sure how much of the code is accessing HTTP info from the context.

FWIW, the approach I implemented is a pretty common pattern I've seen in HTTP server frameworks -- one of the first handlers in the request processing pipeline is dedicated to deserialization of the request body and making it available in a convenient format for the rest of the pipeline. The only complication here is the fact that it has to be statically typed so that just makes the implementation less ergonomic.